src/firewall/firewallService.ts

```diff
@@ -0,0 +1,120 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import { Account, CreateFirewallRuleRequest, HandleFirewallRuleRequest,
+    HandleFirewallRuleParams, HandleFirewallRuleResponse,
+    CreateFirewallRuleResponse,
+    AccountKey,
+    CreateFirewallRuleParams} from '../models/contracts/firewall/firewallRequest';
+import SqlToolsServiceClient from '../languageservice/serviceclient';
+import Constants = require('../constants/constants');
+import { Deferred } from '../protocol';
+import VscodeWrapper from '../controllers/vscodeWrapper';
+
+export class FirewallService {
+
+    private _isSignedIn: boolean = false;
+    private _session: any = undefined;
+    private _account: Account = undefined;
+    private _token = undefined;
+    private _isStale: boolean;
+
+
+    constructor(
+        private _client: SqlToolsServiceClient,
+        private _vscodeWrapper: VscodeWrapper
+    ) {}
+
+    public get isSignedIn(): boolean {
+        return this._isSignedIn;
+    }
+
+    public get account(): Account {
+        return this._account;
+    }
+
+    private convertToAzureAccount(azureSession: any): Account {
+        let tenant = {
+            displayName: Constants.tenantDisplayName,
+            id: this._session.tenantId,
```

Comment:
>_session [](start = 21, length = 8)

You're already passing in the session - it would be better to either just use azureSession or this._session throughout this function